src/server/repository/public/ddl/getCreatePublicSchemaDDL.ts

```diff
@@ -0,0 +1,162 @@
+        on update cascade on delete cascade
+    );
+    create unique index if not exists assessment_cycle_uuid_key on ${schemaName}.assessment_cycle using btree (uuid);
+
+    create table if not exists ${schemaName}.country (
+      country_iso character varying(3) primary key not null,
+      config jsonb not null default '{}'::jsonb
+    );
+
+    create table if not exists ${schemaName}.file (
+      id bigserial primary key,
+      uuid uuid not null default uuid_generate_v4(),
+      name character varying(255) not null,
+      file bytea not null, -- TODO: Remove this when S3 branch merged
+      created_at timestamp without time zone not null default now()
+    );
+    create unique index if not exists file_uuid_key on ${schemaName}.file using btree (uuid);
+
+    create table if not exists ${schemaName}.users (
+      id bigserial primary key,
+      status ${schemaName}.users_status not null default 'invitationPending'::${schemaName}.users_status,
+      email character varying(255) not null,
+      props jsonb not null default '{}'::jsonb,
+      uuid uuid default uuid_generate_v4(),
+      profile_picture_file_uuid uuid,
+      foreign key (profile_picture_file_uuid) references ${schemaName}.file (uuid)
+        on update no action on delete no action
+    );
+    create unique index if not exists users_email_key on ${schemaName}.users using btree (email);
+
+    create table if not exists ${schemaName}.activity_log (
+      id bigserial primary key,
+      time timestamp without time zone not null default timezone('UTC'::text, now()),
+      message text not null,
+      country_iso character varying(3),
+      section character varying(250) not null,
+      target jsonb,
+      user_id bigint not null,
+      assessment_uuid uuid,
+      cycle_uuid uuid,
+      foreign key (assessment_uuid) references ${schemaName}.assessment (uuid)
+        on update cascade on delete cascade,
+      foreign key (country_iso) references ${schemaName}.country (country_iso)
+        on update no action on delete no action,
+      foreign key (cycle_uuid) references ${schemaName}.assessment_cycle (uuid)
+        on update cascade on delete cascade,
+      foreign key (user_id) references ${schemaName}.users (id)
+        on update cascade on delete cascade
+    );
+    create index if not exists idx_activity_log_cycle_message on ${schemaName}.activity_log using btree (cycle_uuid, message);
+    create index if not exists idx_activity_log_filtering on ${schemaName}.activity_log using btree (assessment_uuid, cycle_uuid, country_iso, message, time);
+
+    create table if not exists ${schemaName}.migration_steps (
+      id serial primary key,
+      name character varying(255) not null unique,
+      run_on timestamp without time zone not null default now()
+    );
+
+    -- This table is created separately, but left here for consistency
+    create table if not exists ${schemaName}.migrations (
```

Comment:
Same as above

Not sure if this should be here, or should the script check and create if not exists?


They seem like helper tables that are not directly related to our application
(I would even introduce some separate schema or something to separate them more)

Reply:
For me, it would make sense, let me know

Reply:
My question is, what would be logical place to create the `migrations.public` and `migration.step` tables?

When running the first migration of the migration script, we don't have this table on an empty database and it relies on it.

So I would suggest always running `create table if not exists` before running the script, as it is ran quite rarely in the end, so it doesn't matter to check

Let me know

Reply:
I am referring to this way of doing it:
https://github.com/openforis/fra-platform/pull/3904/files#diff-c0946e94fffdfd9287c7d54e5dd3cb24cfff140c35caef52ae54652767ceeff3R17

Reply:
Yes, it is currently (this pr) implemented as so, I will update the schema and schema creation in the creation of table
👍